Python/sysmodule.c

```diff
@@ -1145,7 +1145,7 @@ sys_getwindowsversion_impl(PyObject *module)
     PyStructSequence_SET_ITEM(version, pos++, PyLong_FromLong(ver.dwMinorVersion));
     PyStructSequence_SET_ITEM(version, pos++, PyLong_FromLong(ver.dwBuildNumber));
     PyStructSequence_SET_ITEM(version, pos++, PyLong_FromLong(ver.dwPlatformId));
-    PyStructSequence_SET_ITEM(version, pos++, PyUnicode_FromString(ver.szCSDVersion));
+    PyStructSequence_SET_ITEM(version, pos++, PyUnicode_FromUnicode(ver.szCSDVersion, wcslen(ver.szCSDVersion)));
```

Comment:
This should be `PyUnicode_FromWideChar` and you can pass `-1` instead of the length.

Reply:
Updated in new commit. Thanks.